nntrainer/tensor/half_tensor.h

```diff
@@ -116,6 +116,37 @@ class HalfTensor : public TensorBase {
    */
   void setZero() override;
 
+  /**
+   * @brief Set the Dist object
+   * @param dist distribution engine
+   */
+  template <typename Engine> void setDist(Engine dist) {
+    NNTR_THROW_IF(!contiguous, std::invalid_argument)
+      // << getName() << " Tensor is not contiguous, cannot set distribution";
+      << " Tensor is not contiguous, cannot set distribution";
+
+    _FP16 *data_ = (_FP16 *)getData();
```

Comment:
How are you planning to connect current TensorV2 with `blas_interface.h` ?
Would it have general function (e.g. gemv ) for `tensor_v2` or separate function (e.g. sgemv/hgemv ) for each `float_tensor`/`half_tensor`? Think I need to refactorize blas interface as well since new `getData` returns void pointer

Reply:
both options would be valid. 
1. we could use the current form since we can manually cast it into a float of fp16 pointer. 
2. we can use different functions depending on the tensor class.